### PR TITLE
[api] support generic github issues

### PR DESCRIPTION
**Pull request by adrianschroeter**
write as

 gh#$project/$repo#$issue

eg

 gh#openSUSE/build#123

needs unfortunality a special hack, since github URLs expect /issues/
inside.

### CODE REVIEW

File: src/api/db/migrate/20190130115727_add_github_issue_tracker.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class AddGithubIssueTracker < ActiveRecord::Migration[5.1]
```
Comment (hennevogel) [inline src/api/db/migrate/20190130115727_add_github_issue_tracker.rb]: This is a data migration and belongs to `db/data`

File: src/api/app/models/issue_tracker.rb [hunk 2]
```diff
@@ -52,6 +52,9 @@ def valid_issue_name?(name)
   def show_url_for(issue, html = nil)
     return unless issue
     url = show_url.gsub('@@@', issue)
+    # generic github tracker needs special handling since they inject '/issues/' inside
+    # github handles issue/pull redirection on it's own luckily
+    url.gsub!(/\#/, '/issues/') if kind == 'github'
```
Comment (Ana06) [inline src/api/app/models/issue_tracker.rb]: Why need this to be a reg exp? :confused:

**Comment by adrianschroeter** [comment]
On Sonntag, 3. Februar 2019, 10:11:24 CET Ana María Martínez Gómez wrote:
> Ana06 commented on this pull request.
> 
> > @@ -52,6 +52,9 @@ def valid_issue_name?(name)
> 
>    def show_url_for(issue, html = nil)
>      return unless issue
>      url = show_url.gsub('@@@', issue)
> +    # generic github tracker needs special handling since they inject
> '/issues/' inside +    # github handles issue/pull redirection on it's own
> luckily
> +    url.gsub!(/\#/, '/issues/') if kind == 'github'
> 
> Why need this to be a reg exp? :confused:

what is the alternative to replace the hash?

-- 

Adrian Schroeter
SUSE Linux Products GmbH,  Maxfeldstr. 5, 90409 Nuernberg, Germany
email: adrian@suse.de

**Comment by coolo** [comment]
> what is the alternative to replace the hash?

If you pass a string, it's automatically converted to regexp. Aka `gsub('#', ...`

**Comment by adrianschroeter** [comment]
On Sonntag, 3. Februar 2019, 13:18:12 CET Stephan Kulow wrote:
> > what is the alternative to replace the hash?
> 
> If you pass a string, it's automatically converted to regexp. Aka `gsub('#',
> ...`

okay, but does it make a difference?

-- 

Adrian Schroeter
SUSE Linux Products GmbH,  Maxfeldstr. 5, 90409 Nuernberg, Germany
email: adrian@suse.de

**Comment by coolo** [comment]
> okay, but does it make a difference?

Well, it's a little easier on the eyes - but other than that: no

File: src/api/db/structure.sql [hunk 3]
```diff
@@ -1435,6 +1435,7 @@ INSERT INTO `schema_migrations` (version) VALUES
 ('20181113095753'),
 ('20181201065026'),
 ('20190111130416'),
-('20190115131711');
+('20190115131711'),
+('20190130115727');
```
Comment (hennevogel) [inline src/api/db/structure.sql]: Data migrations do not write into the `schema_migrations` table, they write to `data_migrations` (which we do not track in the schema as it's instance specific). Just get rid of this chunk.

**Comment by hennevogel** [comment]
@adrianschroeter are you going to move this to `db/data` or what?

**Comment by Ana06** [comment]
what is the state of this? :thinking:

File: src/api/db/migrate/20190130115727_add_github_issue_tracker.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/db/migrate/20190130115727_add_github_issue_tracker.rb]: Moved it...

File: src/api/app/models/issue_tracker.rb (hunk 2 above)
Comment (hennevogel) [reply src/api/app/models/issue_tracker.rb]: Changed to `'#'`

**Pull request merged**